projects/components/src/overlay/modal/modal-overlay.module.ts

```diff
@@ -3,8 +3,10 @@ import { NgModule } from '@angular/core';
 import { ButtonModule } from '../../button/button.module';
 import { TooltipModule } from '../../tooltip/tooltip.module';
 import { ModalOverlayComponent } from './modal-overlay.component';
+import { ModalService } from './modal-service';
 
 @NgModule({
+  providers: [ModalService],
```

Comment:
providers should either belong to a component or be in root. Declaring them in the module is effectively root, but is more confusing to the reader.